oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/stream/BanyanDBStorageDAO.java

```diff
@@ -0,0 +1,140 @@
+import org.apache.skywalking.oap.server.core.storage.AbstractDAO;
+import org.apache.skywalking.oap.server.core.storage.IManagementDAO;
+import org.apache.skywalking.oap.server.core.storage.IMetricsDAO;
+import org.apache.skywalking.oap.server.core.storage.INoneStreamDAO;
+import org.apache.skywalking.oap.server.core.storage.IRecordDAO;
+import org.apache.skywalking.oap.server.core.storage.StorageDAO;
+import org.apache.skywalking.oap.server.core.storage.type.StorageBuilder;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.BanyanDBStorageClient;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.schema.AlarmRecordBuilder;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.schema.BrowserErrorLogRecordBuilder;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.schema.EventBuilder;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.schema.LogRecordBuilder;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.schema.Metadata;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.schema.NetworkAddressAliasBuilder;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.schema.ProfileTaskLogRecordBuilder;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.schema.ProfileTaskRecordBuilder;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.schema.ProfileThreadSnapshotRecordBuilder;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.schema.SegmentRecordBuilder;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.schema.UITemplateBuilder;
+
+import java.util.Map;
+
+@Slf4j
+public class BanyanDBStorageDAO extends AbstractDAO<BanyanDBStorageClient> implements StorageDAO {
+    public BanyanDBStorageDAO(BanyanDBStorageClient client) {
+        super(client);
+    }
+
+    @Override
+    public IMetricsDAO newMetricsDao(StorageBuilder storageBuilder) {
+        try {
+            Class<?> returnType = storageBuilder.getClass().getMethod("storage2Entity", Map.class).getReturnType();
+            if (Event.class.equals(returnType)) {
+                return new BanyanDBMetricsDAO<>(new EventBuilder());
+            } else if (ServiceTraffic.class.equals(returnType)) {
+                return new BanyanDBMetricsDAO<>(new Metadata.ServiceTrafficBuilder());
+            } else if (InstanceTraffic.class.equals(returnType)) {
+                return new BanyanDBMetricsDAO<>(new Metadata.InstanceTrafficBuilder());
+            } else if (EndpointTraffic.class.equals(returnType)) {
+                return new BanyanDBMetricsDAO<>(new Metadata.EndpointTrafficBuilder());
+            } else if (NetworkAddressAlias.class.equals(returnType)) {
+                return new BanyanDBMetricsDAO<>(new NetworkAddressAliasBuilder());
+            } else {
+                throw new IllegalStateException("record type is not supported");
+            }
+        } catch (NoSuchMethodException noSuchMethodException) {
```

Comment:
MetricsDao with metadata only? Are you sure this is the right way to do this?
`StorageBuilderFactory#Default` bases on the class scan mechanism. If you want your own, you should create another way, such as codes or annotation to declare this. But never do this kind of reflect and `if/else`

Reply:
```java
@Stream(name = ServiceTraffic.INDEX_NAME, scopeId = DefaultScopeDefine.SERVICE,
    builder = ServiceTraffic.Builder.class, processor = MetricsStreamProcessor.class)
@MetricsExtension(supportDownSampling = false, supportUpdate = false)
@EqualsAndHashCode(of = {
    "name",
    "nodeType"
})
public class ServiceTraffic extends Metrics {
```

This is how Builder is declared and mapped to the entity. I think you should have a class or a class with annotations to declare the override? Such as(only an example)
```java
@StreamBuilderOverride(name = ServiceTraffic.INDEX_NAME,  builder = BanyanDBServiceTrafficBuilder.class)
@StreamBuilderOverride(name = InstanceTraffic.INDEX_NAME,  builder = BanyanDBInstanceTrafficBuilder.class)
public class BanyanDBStorageBuilderFactory
```

Reply:
I missed this part. I will check it.